restful_api/controllers/token.py

```diff
@@ -0,0 +1,110 @@
+
+#--------------------------------- Access_Key Login url ----------------------------------------#
+
+class AccessToken(http.Controller):
+    """."""
+
+    def __init__(self):
+
+        self._token = request.env["api.access_token"]
+        self._expires_in = request.env.ref(expires_in).sudo().value
+
+    @http.route("/api/login", methods=["POST","OPTIONS"], type="http", auth="none", csrf=False, cors='*')
+    def token(self, **post):
+        input_user_name = post.get("username"),
+        input_password = post.get("password"),
+
+        if input_user_name[0] != None:
+
+            res_usersss = request.env['res.users'].sudo().search([
+                ('active','=',True),
+                ('login','=',input_user_name)])
+            
+            
+            if res_usersss:
+                request.env.company_id = 1  ###set company_id based logic users 
+                exp = datetime.datetime.utcnow() + datetime.timedelta(days=30)
+
+                rbytes = os.urandom(64)
+                random_pass = str(hashlib.sha256(rbytes).hexdigest())
+
+                token_values = request.env['api.access_token'].sudo().create({
+                            'user_id': res_usersss.id,
+                            'expires': exp.strftime(DEFAULT_SERVER_DATETIME_FORMAT),
+                            'token': str(random_pass)})
+
+                token_key_visible = {
+                                        "status":200,
+                                        "message":"Success",
+                                        "data":[
+                                            {
+                                                    "user_id": res_usersss.id,    
+                                                    "username": input_user_name[0],
+                                                    "access_token": str(random_pass),
+                                            }]
+                                    }
+            
+                
+                
+                return werkzeug.wrappers.Response(
+                status=200,
+                content_type="application/json; charset=utf-8",
+                headers=[("Cache-Control", "no-store"),('Access-Control-Allow-Headers', 'Origin, Content-Type, X-Auth-Token, charset'),('Access-Control-Allow-Methods','POST, GET, OPTIONS, DELETE, PUT'), ('Access-Control-Allow-Origin', '*'), ("Pragma", "no-cache")],
+                response=json.dumps(token_key_visible, default=default),)
+        
+            else:
+                return werkzeug.wrappers.Response(
+                    status=401,
+                    content_type="application/json; charset=utf-8",
+                    headers=[("Cache-Control", "no-store"),('Access-Control-Allow-Headers', 'Origin, Content-Type, X-Auth-Token, charset'),('Access-Control-Allow-Methods','POST, GET, OPTIONS, DELETE, PUT'), ('Access-Control-Allow-Origin', 'http://localhost:8069'), ("Pragma", "no-cache")],
```

Comment:
localhost is hardcoded here. Will this case issues in case when people access odoo in non-localhost mode, remotely.

Reply:
No issues will occur with this, it is just a responce message, now the code was removed.